src/Raven.Server/Documents/Indexes/Static/Roslyn/Rewriters/CaptureSelectNewFieldNamesVisitor.cs

```diff
@@ -137,5 +137,11 @@ private static void ThrowIndexingFunctionMustReturnAnonymousObjectOrDictionary(b
 
             throw new InvalidOperationException(message);
         }
+
+        public void Reset()
+        {
+            Fields = null;
+            _isQueryExpression = new bool?();
```

Comment:
`= null` ?

Reply:
We should reset the internal state of the Nullable<bool>